### ISSUE TITLE
clonalDiversity gives different results when using group.by = "ID"

### ISSUE THREAD

**Issue by gilstel** [ncborcherding/scRepertoire#148]
We performed the following steps - 
```

contig.list.bcr.alz = list(dCLNs.wt, dCLNs.ad, sCLNs.wt, sCLNs.ad, iLNs.wt, iLNs.ad)

# "samples" and "ID" must be different strings
combined.bcr = combineBCR(contig.list.bcr.alz, 
                          samples = c("dCLNs.wt", "dCLNs.ad", "sCLNs.wt", "sCLNs.ad", "iLNs.wt","iLNs.ad"), 
                          ID = c("dCLNs_wt", "dCLNs_ad", "sCLNs_wt", "sCLNs_ad", "iLNs_wt", "iLNs_ad"))

gg = clonalDiversity(df = combined.bcr, cloneCall = "gene", n.boots = 100)

![clonalDiversity](https://user-images.githubusercontent.com/50887514/158958549-7b8b4586-f0d6-476a-8373-19fc6c660b0b.jpg)

gg1 = clonalDiversity(df = combined.bcr, cloneCall = "gene", group.by = "ID", n.boots = 100)

![clonalDiversityGroupById](https://user-images.githubusercontent.com/50887514/158958568-90a1abaf-1fe4-4b25-9137-5fd0168e30bd.jpg)

We get completely different results.  
When we do not use group.by, the samples and ID strings are concatenated and dCLN.wt_dCLN_wt (sample_ID) appears on the bottom whereas when using group.id = "ID" 
dCLN_ad appears on the bottom.  We would have expected the same sample to appear in the same place since the cells in question are the same, irrespective of using implicit "sample_ID" or "ID".  What could be the reason for this and which plot should we believe?

We are probably missing something basic.  
Any imput would be much appreciated.
```

We could not perform combineTCR without entering IDs and they had to be different from the sample strings.

Once

**Comment by ncborcherding** [comment]
Hey glistel, 

Thanks for reaching out! Could you let me know what version of scRepertoire you have installed?

`clonaDiversity()` randomly takes a sample of clones using the smallest sequencing run/library as the size reference and does it 100 times. So there should be some difference even between the same called of `clonalDiversity()`.  This difference can be quite sporadic if there is a sample with very small number of clonotypes. What is the relative size of your BCR runs?  You can get this with just simply:

```
lapply(combined.bcr, dim)
```

I don't think that is necessarily the issue - the metric numbers would be lower across the board. Looking closely at the results you are getting similar sample placement for each metric, but the **Group** is different :

![Screen Shot 2022-03-18 at 4 45 24 AM](https://user-images.githubusercontent.com/22754118/158979692-28defb5e-45b7-4806-8a37-b5199a0134af.png)

To me this indicates there is likely issue with the code when it comes to *group.by* = "ID. Would you mind [emailing me](mailto:ncborch@gmail.com) your combine.bcr object so I can reproduce the issue and get a fix committed to github? 

> We could not perform combineTCR without entering IDs and they had to be different from the sample strings.

What version of scRepetoire are you using? Any of the github repo versions > 1.3 should not require IDs anymore. The bioconducto version 1.5.1 also does not require this. ID strings do not have to be different than one another, but need to match the same length as the sample string, so for example using the ```combineBCR()```:

```
combined.bcr = combineBCR(contig.list.bcr.alz, 
                          samples = c("dCLNs", "dCLNs", "sCLNs", "sCLNs", "iLNs","iLNs"), 
                          ID = c("wt", "ad", "wt", "ad", "wt", "ad"))
```

Thanks,
Nick

**Comment by gilstel** [comment]
Hi Nick

Tried emailing you but got no response
Tried another email (borcherding.n@email.wustl.edu)

Thanks,
Gil

**Comment by ncborcherding** [comment]
Hey Gil,

Apologies missed your first two emails for some reason on the gmail account but got yours earlier today. I did find the issue and commited the changes to the dev branch of the github repo. Here are the following results I got after the change. 

![Screen Shot 2022-03-24 at 10 25 43 AM](https://user-images.githubusercontent.com/22754118/159951742-bc2a106c-b01d-41df-8695-2a36fb1e76af.png)


![Screen Shot 2022-03-24 at 10 25 47 AM](https://user-images.githubusercontent.com/22754118/159951817-bdda8a92-0eef-49cf-8b6e-a52e4343541d.png)

To instal use the following code and make sure you restart your R session.

```
devtools::install_github("ncborcherding/scRepertoire@dev")
```

Thanks so much for sending the data and being persistent!!

Nick
